apps/rxjs-pipe-bug/src/app/app.service.ts

```diff
@@ -8,15 +8,13 @@ export class AppService {
 
   getAll$ = this.dbService.infos$;
 
+  // prettier-ignore
   deleteOldTopics(type: TopicType): Observable<boolean> {
     return this.dbService.searchByType(type).pipe(
       take(1),
-      mergeMap((topicToDelete) =>
-        topicToDelete.length > 0
-          ? topicToDelete
-              .map((t) => this.dbService.deleteOneTopic(t.id))
-              .reduce((acc, curr) => merge(acc, curr), of(true))
-          : of(true)
+      switchMap(topicsToDelete =>
+        forkJoin(topicsToDelete.map((t) => this.dbService.deleteOneTopic(t.id))).pipe(
+          map(results => !results.some(result => !result)))
```

Comment:
you can also use the `every` operator to avoid having a double negative condition

Reply:
yes, meanwhile I found the docs that `every `on an empty array works as needed here.